### PR TITLE
[query] Graphite response time ranges are end-time based (not start-time based)

### PR DESCRIPTION
**Pull request by teddywahle**
**What this PR does / why we need it**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing and/or backwards incompatible change?**:

```release-note

```

**Does this PR require updating code package or user-facing documentation?**:

```documentation-note

```

### CODE REVIEW

File: src/query/graphite/storage/m3_wrapper.go [hunk 1]
```diff
@@ -264,5 +267,11 @@ func (s *m3WrappedStore) FetchByQuery(
 		return nil, err
 	}
 
+	fmt.Println("TRANSLATED SERIES")
+	if len(series) > 0 {
+		fmt.Println(series[0].StartTime(), series[0].EndTime())
+		fmt.Println(series[0].SafeValues())
+	}
+
```
Comment (teddywahle) [inline src/query/graphite/storage/m3_wrapper.go:275]:
```suggestion
```

File: src/query/graphite/storage/m3_wrapper.go [hunk 2]
```diff
@@ -134,17 +134,18 @@ func truncateBoundsToResolution(
...
 	steps := math.Floor(length / float64(resolution))
+
 	truncatedLength := time.Duration(steps) * resolution
-	end = start.Add(truncatedLength)
-	return start, end
+	truncatedStart := truncatedEnd.Add(truncatedLength * -1)
+
+	// start one step earlier, unless the start time was already truncated
+	if !truncatedStart.Equal(start) {
+		truncatedStart = truncatedStart.Add(resolution * -1)
```
Comment (yyinsc) [inline src/query/graphite/storage/m3_wrapper.go:146]: discussed offline, this special handling seems a bit of intuitive. needs to be changed.

Comment (teddywahle) [reply src/query/graphite/storage/m3_wrapper.go]: Removed

File: src/query/graphite/storage/m3_wrapper.go [hunk 3]
```diff
@@ -178,31 +174,38 @@ func translateTimeseries(
...
 		for _, datapoint := range dps.Datapoints() {
 			ts := datapoint.Timestamp
-			if ts.Before(start) {
+			if ts.Before(seriesStartTime) {
 				// Outside of range requested.
 				continue
 			}
 
-			if !ts.Before(end) {
+			if !(ts.Before(seriesEndTime) || ts.Equal(seriesEndTime)) {
```
Comment (yyinsc) [inline src/query/graphite/storage/m3_wrapper.go:198]: nit: change to `ts.After(seriesEndTime)`

**Comment by robskillington** [comment]
We are hoping to keep the default implementation the same since folks in production rely on the current windowing. Can you confirm/deny if using new config options from PR https://github.com/m3db/m3/pull/2716 helps match the windowing you're trying to match?

**Pull request closed without merging**